resolwe/observers/decorators.py

```diff
@@ -0,0 +1,37 @@
+"""Decorators for observers."""
+
+
+from resolwe.observers.models import Observer
+from resolwe.observers.protocol import post_container_changed, pre_container_changed
+
+
+def move_to_container(function):
+    """Indicate object is moving between containers.
+
+    The notifications are suppressed for the first argument (usually self) of the given
+    function which must be the observable object.
+    """
+
+    def wrapper(self, *args, **kwargs):
+        pre_container_changed.send(sender=type(self), instance=self)
+        suppress_observer_notifications(function)(self, *args, **kwargs)
+        post_container_changed.send(sender=type(self), instance=self)
+
+    return wrapper
+
+
+def suppress_observer_notifications(function):
+    """Suppress the observer notifications.
+
+    The notifications are suppressed for the first argument (usually self) of the given
+    function which must be the observable object.
```

Comment:
Maybe we should check that self is an instance of Observable?

Reply:
Actually, we have to check that it is an instance of `PermissionObject`. 

Maybe it would be nicer to add the `Observable` model to observer models and check for `Observable`  to increase code readability? 

It could be just an alias for the `PermissionModel`.